src/VisualStudio/Next/Remote/ServiceHubRemoteHostClient.JsonRpcSession.cs

```diff
@@ -160,6 +154,62 @@ private class SnapshotJsonRpcClient : JsonRpcClient
                     }
                 }
 
+                private async Task WriteAssetAsync(ObjectWriter writer, byte[][] checksums)
+                {
+                    // special case
+                    if (checksums.Length == 0)
+                    {
+                        await WriteNoAssetAsync(writer).ConfigureAwait(false);
+                        return;
+                    }
+
+                    if (checksums.Length == 1)
+                    {
+                        await WriteOneAssetAsync(writer, checksums[0]).ConfigureAwait(false);
+                        return;
+                    }
+
+                    await WriteMultipleAssetsAsync(writer, checksums).ConfigureAwait(false);
+                }
+
+                private Task WriteNoAssetAsync(ObjectWriter writer)
```

Comment:
all three of these look practically identical.  a  count, followed by the items.  WHy do we need to special case 0 and 1 don't they fall out naturally from calling the many case?